org-code-javabuilder/lib/src/main/java/org/code/javabuilder/CodeBuilderWrapper.java

```diff
@@ -0,0 +1,66 @@
+package org.code.javabuilder;
+
+import static org.code.protocol.LoggerNames.MAIN_LOGGER;
+
+import java.util.logging.Logger;
+import org.code.protocol.*;
+import org.json.JSONObject;
+
+/**
+ * This creates a high-level wrapper that handles errors and exceptions in a way that can be
+ * communicated to the user and logged by us. Note: All exceptions, runtime exceptions, and errors
+ * are caught. This allows us to log our errors and communicate failure states to the user. This may
+ * be thought of as similar to an HTTP error handler.
+ */
+public class CodeBuilderWrapper {
+  private final ProjectFileLoader fileLoader;
+  private final OutputAdapter outputAdapter;
+
+  public CodeBuilderWrapper(ProjectFileLoader fileLoader, OutputAdapter outputAdapter) {
+    this.fileLoader = fileLoader;
+    this.outputAdapter = outputAdapter;
+  }
+
+  public void executeCodeBuilder() {
+    try {
+      UserProjectFiles userProjectFiles = fileLoader.loadFiles();
+      try (CodeBuilder codeBuilder =
+          new CodeBuilder(GlobalProtocol.getInstance(), userProjectFiles)) {
+        codeBuilder.buildUserCode();
+        codeBuilder.runUserCode();
+      }
+    } catch (InternalServerError | InternalServerRuntimeError e) {
+      // The error was caused by us (essentially an HTTP 5xx error). Log it so we can fix it.
+      JSONObject eventData = new JSONObject();
+      eventData.put("exceptionMessage", e.getExceptionMessage());
+      eventData.put("loggingString", e.getLoggingString());
+      Logger.getLogger(MAIN_LOGGER).severe(eventData.toString());
+
+      // The error affected the user. Tell them about it.
+      outputAdapter.sendMessage(e.getExceptionMessage());
+    } catch (JavabuilderException | JavabuilderRuntimeException e) {
+      // The error affected the user and was caused by them. Tell them about it.
+      outputAdapter.sendMessage(e.getExceptionMessage());
```

Comment:
Not a huge deal, but would it be conceptually clearer to just have the catch block for `JavabuilderException | JavabuilderRuntimeException` but then only log if the exception is an instance of InternalServer(Runtime)Error? Might be easier to follow the hierarchy since InternalServer(Runtime)Errors are Javabuilder(Runtime)Exceptions

Reply:
That's a good idea. I'm going to merge as-is and then I may do a follow-up PR.